src/parser/extract-stories.ts

```diff
@@ -0,0 +1,83 @@
+import dedent from 'ts-dedent';
+import * as svelte from 'svelte/compiler';
+import { extractId } from './extract-id';
+
+interface StoryDef {
+  name: string;
+  template: boolean;
+  source: string;
+  hasArgs: boolean;
+}
+
+function getStaticAttribute(name: string, node: any): string {
+  // extract the attribute
+  const attribute = node.attributes.find(
+    (att: any) => att.type === 'Attribute' && att.name === name
+  );
+
+  if (!attribute) {
+    return null;
+  }
+
+  const { value } = attribute;
+  // expect the attribute to be static, ie only one Text node
+  if (value && value.length === 1 && value[0].type === 'Text') {
+    return value[0].data;
+  }
+
+  throw new Error(`Attribute ${name} is not static`);
+}
+
+/**
+ * Parse a Svelte component and extract stories.
+ * @param component Component Source
+ * @returns Map of storyName -> source
+ */
+export function extractStories(component: string): Record<string, StoryDef> {
+  // compile
+  const { ast } = svelte.compile(component);
+
+  const stories: Record<string, StoryDef> = {};
+  svelte.walk(ast.html, {
+    enter(node: any) {
+      if (node.type === 'InlineComponent' && (node.name === 'Story' || node.name === 'Template')) {
```

Comment:
What happens if someone creates their own component named `Story` or `Template`? Will those match here?

Reply:
You're right, If a user write : 

```html
<script>
  import { Story as SBStory } from '@storybook/addon-svelte-csf'
  import Story from './Story.svelte'
</script>
``` 

then this check will fail. I can try to parse the import statements in the script tag.